tests/rados/test_osd_ecpool_inconsistency_scenario.py

```diff
@@ -154,56 +161,183 @@ def run(ceph_cluster, **kw):
         log.info(
             f"The osd_scrub_auto_repair_num_errors value is set to {auto_repair_param_value}"
         )
-        obj_count = get_inconsistent_count(scrub_object, pg_id, rados_obj, "scrub")
+        try:
+            obj_count = get_inconsistent_count(scrub_object, pg_id, rados_obj, "scrub")
+        except Exception as e:
+            log.info(e)
         if obj_count == -1:
             log.error(f"The scrub not initiated on the pg-{pg_id}")
             rados_obj.log_cluster_health()
             return 1
-
-        if obj_count != 0:
+        if obj_count != no_of_objects:
             log.error(
                 f"Scrub repaired the {no_of_objects - obj_count} inconsistent objects"
             )
-            rados_obj.log_cluster_health()
-            return 1
-        result = verify_pg_state(rados_obj, pg_id)
-        if not result:
-            log.error("The  pg state output contain repair state after scrub")
-            rados_obj.log_cluster_health()
             return 1
+        log.info(
+            "Test scenario2 completed: SCRUB-Inconsistent objects less than the osd_scrub_auto_repair_num_errors "
+            "count and osd_scrub_auto_repair is false"
+        )
+        log.info(
+            "Test scenario3: SCRUB-Inconsistent objects less than the osd_scrub_auto_repair_num_errors count and "
+            "osd_scrub_auto_repair is true"
+        )
+        mon_obj.set_config(section="osd", name="osd_scrub_auto_repair", value="true")
 
+        # Case3: The inconsistent object count is less than osd_scrub_auto_repair_num_errors count and
+        # osd_scrub_auto_repair is True
+        # # Commented the code due to the BZ#2335762- Scrub operations fail to repair inconsistent data in the EC pool
+        # # when the nodeep-scrub flag is enabled. To track the issue jira task is -RHCEPHQE-17391
+        #
+        # obj_count = get_inconsistent_count(scrub_object, pg_id, rados_obj, "scrub")
+        #
+        scrub_obj.set_osd_flags("unset", "nodeep-scrub")
```

Comment:
move the unseting of deep-scrub just before deleting the pools.

Reply:
Done.